### DISCUSSION TITLE
Search using gID from title use custom regular expressions

### DISCUSSION THREAD

**Discussion by zizzdog** [Difegue/LANraragi#925] [Q&A] [answered]
I found that the original plugin can only search for numbers inside square brackets [xxx] as GID, but the GID of the files I downloaded start with a number. 
Although I can read and modify the code to suit my needs, could you please explain the matching rules on the plugin page? It would be even better if there could set a regular expression input box to accommodate everyone's needs.

I usually use EhViewer to download files, and the file names are typically like this:
123456-(C100) [Group (Name)] Title [Language].zip

**Comment by polak14** [comment]
That's the way h@h downloader names archives, and generally the most used format (e.g. chaika downloads or pandaforever by default)

**Comment by Difegue** [comment]
I really don't want to expose the regex since it's atrociously complex and people will mess it up, but the source code for filename parsing details it fully:  

https://github.com/Difegue/LANraragi/blob/f52b9904d9389e6e3ec13cae81f60bd6156d5026/lib/LANraragi/Plugin/Metadata/RegexParse.pm#L109

**Comment by Difegue** [comment, marked as answer]
Ah my bad you're asking about the EH plugin -- I'd recommend throwing the regex into something like regex101 to experiment, but basically `/\[([0-9]+)\]/g` just checks if the name starts with a bracketed GID.  

I guess for you something like `/([0-9]+)-/g` would work.

**Reply by zizzdog** [reply]
Yes, I modified it like that. and now we have GPT to explain these complex and difficult-to-understand expressions.🙃
